### ISSUE TITLE
Dragon Force Fails to extract

### ISSUE THREAD

**Issue by whitezombie2000** [slinga-homebrew/Save-Game-Extractor#3]
I'm getting a message "What the heck too small: 18058" every time I try to extract it. It's a bigger save file and takes about ten minutes. 

https://imgur.com/a/p2MhvuB

**Comment by slinga-homebrew** [comment]
can you upload the save here?

**Comment by whitezombie2000** [comment]
I can't upload a save that I tried to dumped because after it fails its 0 bytes.  I can provide a save dumped from the Saturn Gamer's Cartridge.

[DR_FORCE_00.zip](https://github.com/slinga-homebrew/Save-Game-Extractor/files/5302849/DR_FORCE_00.zip)

**Comment by whitezombie2000** [comment]
I have three saves on my external. Only save DR_FORCE_00 the first save is failing to extract. DR_FORCE_01 and DR_FORCE_02 extracted fine from the external. I moved each save to internal one by one, its a big save, the internal can only keep one, and got the same results trying to extract. DR_FORCE_00 fails to extract from the internal, but DR_FORCE_01 and DR_FORCE_02 extract fine.

**Comment by slinga-homebrew** [comment]
I made a Dragon Force save and unfortunately it wasn't the same size as yours after compression. I'm not sure what the issue is but I made a debug ISO for you to try: https://github.com/slinga-homebrew/Save-Game-Extractor/releases/tag/0.96.2. 

Let me know if that fixes your issue. I can test with the save you supplied as well.

**Comment by whitezombie2000** [comment]
I tried your debug ISO but it loads Save Game Copier and not the Extractor. So i'm not sure if that's a mistake or not.  You can try my supplied save if you want to but it doesn't seem to matter. I tried the supplied save, created a new one with Wein, and it fails extracting DR_FORCE_00. Interestingly I tried a new game with Mikhal and Teiris and had no issue with their DR_FORCE_00 file on the internal.  I also tried my Japanese Saturn and produced the same results as my American Saturn.

**Comment by slinga-homebrew** [comment]
Sorry about that. Fail on my part. I reuploaded the ISO:  https://github.com/slinga-homebrew/Save-Game-Extractor/releases/tag/0.96.2. It should says Save Game Extractor Ver 0.97.1 when you boot it.

**Comment by slinga-homebrew** [comment]
What I'm stumped with is why you get a 0-byte file. Even if there is an error later in the transmission you should still have received some of the file.

**Comment by whitezombie2000** [comment]
So can you extract file DR_FORCE_00 with Wein successfully?

**Comment by slinga-homebrew** [comment]
I haven't tried yet. I got the same error message "What the heck too short" but I was testing on an emulator.

**Comment by slinga-homebrew** [comment]
5517 % 128 = 13. So it's likely that for very small sizes I have an issue.

**Comment by whitezombie2000** [comment]
Still fails, What the heck too small: 9770 with the test ISO. Here is the dumped file.

[dftest3.zip](https://github.com/slinga-homebrew/Save-Game-Extractor/files/5476136/dftest3.zip)

**Comment by whitezombie2000** [comment]
Ok even though I got the error, the extraction still worked. I was able successfully copy the the DR_FORCE._00 back to the wiped internal memory and load the save.

@ubuntu3:~/Save-Game-Extractor$ python3 sgex.py dftest3.bin
Save Game Extractor
(github.com/slinga-homebrew/Save-Game-Extractor)

Errors Corrected: 2
Transmitted Filename: DR_FORCE_00
Transmitted Save Size: 18144
Transmitted MD5: 6ad39bd7c2bfa63d0d1556eda830ca50
Computed MD5: 6ad39bd7c2bfa63d0d1556eda830ca50

MD5 hashes validate, save is correct.
Wrote save game DR_FORCE_00 to disk

**Comment by whitezombie2000** [comment]
Ok I went back and tested 0.97 again. I got the what the heck too small error with the first save file like normal but the dumped file was not 0 bytes. So maybe I had some bad transmissions when previously testing with 0.97. Again I was able to successfully run the python script, copy it back over to the Saturn even with the what the heck to small error.

**Comment by slinga-homebrew** [comment]
I thought I removed the what the heck message in the latest build. I'm glad it worked. 

I do think I need to test with small transfer sizes (or small sizes % 128). Let me know if you still have issues.

**Comment by whitezombie2000** [comment]
Is there a way to do this on a laptop via an inexpensive USB adapter with line in or some other method? I ask because most laptops only have microphone and line out. I was using a desktop with the blue line in port but was looking for a more portable method.

**Comment by slinga-homebrew** [comment]
I used this USB adapter on my laptop: https://www.amazon.com/gp/product/B00IRVQ0F8/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1. $8.00

**Comment by whitezombie2000** [comment]
That has line in and it works? I see microphone in and audio out on that USB adapter which most laptops already have. Line in is usually colored blue.

**Comment by slinga-homebrew** [comment]
I didn't know there was a difference between microphone in and line in. I used the microphone in and it worked fine on my laptop.

**Comment by whitezombie2000** [comment]
I tried the microphone in port on a laptop, it failed the "this is cool" test, and it didn't transfer any save data. Maybe that USB adapter has line in which is why it works. Maybe I will order one and try it. If you plug the USB adapter into Windows and go to sound devices, does it show line in?

**Comment by slinga-homebrew** [comment]
I don't have a Windows machine to test on.

**Comment by whitezombie2000** [comment]
I ordered the Sabrent USB adapter and I will let you know if it works when I get it. Are you running Ubuntu?

**Comment by slinga-homebrew** [comment]
I'm on Linux Mint which which is based on Ubuntu.

**Comment by whitezombie2000** [comment]
I finally got the USB adapter and it works great using the pink microphone plugin. Look at the sound devices listed in Ubuntu I see "Internal Microphone - Built-in Audio" listed and the USB adapter "Microphone - Audio Adapter (Unitek Y247A)". I find it very strange that the USB microphone works for this but the built in one on the side of the laptop doesn't. I think I tried the built in pink microphone port on the desktop PC and it didn't work too. Again thanks for your help, this is an awesome tool!
